### PR TITLE
Generate cut handling code only when necessary

### PR DESCRIPTION
**Pull request by MatthieuDartiailh**
I forgot this one last week...

### CODE REVIEW

File: tests/test_pegen.py [hunk 1]
```diff
@@ -532,6 +532,17 @@ def test_cut() -> None:
     ]
 
 
+def test_cut_early_exit() -> None:
```
Comment (pablogsal) [inline tests/test_pegen.py:535]: How does this test check that we are exiting early? Or is just for coverage?

**Review by pablogsal — APPROVED** [review]

File: tests/test_pegen.py (hunk 1 above)
Comment (MatthieuDartiailh) [reply tests/test_pegen.py:535]: If the cut was not honored we would succeed to parse `(a)` as `( NAME )` and the syntax error would not be raised.

Comment (pablogsal) [reply tests/test_pegen.py:535]: Hummm, but the cut is honored even before this PR no? IIUC this is just optimizing when we add the error handling code

Comment (MatthieuDartiailh) [reply tests/test_pegen.py:535]: Yes but at one point I messed up part of the logic and that was not working anymore. I think having a sanity check is a good thing.

**Pull request merged**